src/ImageOverlayDialog.cc

```diff
@@ -38,6 +38,23 @@ ImageOverlayDialog::ImageOverlayDialog(QPixmap image, QWidget *parent)
 	raise();
 }
 
+void ImageOverlayDialog::reject()
+{
+	// needed on macOS to recover the main menu after the dialog is closed(!)
+	setWindowState(Qt::WindowNoState);
+
+	// It might be slightly cleaner to do:
+	//
+	// #ifdef Q_WS_MACX
+	// #include <Carbon/Carbon.h>
```

Comment:
The problem on macOS here is that for whatever reason, when the ImageOverlayDialog is dismissed, the fullscreen state isn't fully reset for the app - meaning you lose the application menu bar for the rest of the app's lifetime.

I assume this is either a bug in Qt where it fails to call the MacOS API `ShowMenuBar()` to reset the UI when closing a QWindow with WindowState `QWindow::FullScreen`, or there's some missing housekeeping that lets Qt reset the app correctly.

I think the second option is more likely, so I'm faking the cleanup by forcing the QDialog to de-fullscreen itself before it gets destroyed, which correctly resets the app's menu bar.  This could also affect other platforms where Qt implements some application-wide behaviour for fullscreen (e.g. hiding the menu bar), so I think this is the better solution than doing the macOS-specific hack of breaking the Qt abstraction and directly calling the Carbon API needed to hack around the problem.

By the way, why does the constructor `raise()`?